### PR TITLE
Fix vacate for integrations-server kind.

### PR DESCRIPTION
**Pull request by gigerdo**
When vacating an integrations-server instance, the automatic tracking of the request currently did not work. This is due to the API lacking support for integrations-server (APM and integrations-server are returned as APM).

The underlying problem is that searching for the resource by using the resource-id + kind (APM) will not find anything.

Since the resource-ids are unique, we can just search by resource-id to find the relevant resource to track.

## Related Issues
Fixes: https://github.com/elastic/cloud-sdk-go/issues/453

## How Has This Been Tested?
Unit-tests & manual tests with ecctl

## Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

### CODE REVIEW

File: pkg/plan/lookup_query.go [hunk 1]
```diff
@@ -36,3 +36,28 @@ func NewReverseLookupQuery(resourceID, kind string) *models.SearchRequest {
 		},
 	}}
 }
+
+// LookupByResourceIdQuery can be used to find a deployment by a resource-id (can be any kind e.g. elasticsearch, kibana, etc.)
+// (Builds a query that searches all possible kinds for the resource-id)
+func LookupByResourceIdQuery(resourceID string) *models.SearchRequest {
+	kinds := []string{"elasticsearch", "kibana", "apm", "appsearch", "enterprise_search", "integrations_server"}
```
Comment (Z3r0Sum) [inline pkg/plan/lookup_query.go:43]: Nit: is there a way we can derive these rather than hardcode to future proof?

**Review by Z3r0Sum — APPROVED** [review]
LGTM - left a nit, unsure if this is possible in an easy fashion.

edit: looking deeper at the code, this also might just be generally confusing how this works imho.  Just following the trail:

- [Get the various types of moves per allocator](https://github.com/elastic/cloud-sdk-go/blob/master/pkg/api/platformapi/allocatorapi/vacate.go#L194-L268) ->
- [Vacate Cluster Parameters](https://github.com/elastic/cloud-sdk-go/blob/master/pkg/api/platformapi/allocatorapi/vacate.go#L270-L293) ->
- [Vacate Cluster](https://github.com/elastic/cloud-sdk-go/blob/master/pkg/api/platformapi/allocatorapi/vacate.go#L309-L318).  This is where as a reader, I'd be confused as we eventually perform validations on a specific list of kinds as seen [here](https://github.com/elastic/cloud-sdk-go/blob/master/pkg/api/platformapi/allocatorapi/vacate_params.go#L44-L50).
- [Move Clusters by Type](https://github.com/elastic/cloud-sdk-go/blob/master/pkg/api/platformapi/allocatorapi/vacate.go#L452-L454) - [params](https://github.com/elastic/cloud-sdk-go/blob/master/pkg/api/platformapi/allocatorapi/vacate.go#L386-L388) && [cluster type set](https://github.com/elastic/cloud-sdk-go/blob/master/pkg/api/platformapi/allocatorapi/vacate.go#L422-L440)

With all this said, aren't we currently just getting lucky that we're getting some form of the APM cluster type back for the underlying API call without addressing it in the core logic?  The tracking updates make sense in essence; however, they contradict how everything in the code gets handle to perform the vacate request.  Curious about your thoughts?  Perhaps this can be done in a better way in general and a follow-up PR can happen afterwards.

**Review by dimuon — APPROVED** [review]
LGTM

File: pkg/plan/lookup_query.go (hunk 1 above)
Comment (gigerdo) [reply pkg/plan/lookup_query.go:43]: There is no list of kinds in the API definition from what I can see, so we can't automatically get it.

**Comment by gigerdo** [comment]
> With all this said, aren't we currently just getting lucky that we're getting some form of the APM cluster type back for the underlying API call without addressing it in the core logic? The tracking updates make sense in essence; however, they contradict how everything in the code gets handled to perform the vacate request. Curious about your thoughts? Perhaps this can be done in a better way in general and a follow-up PR can happen afterwards.

Yes, this is just a workaround. We would also have to extend the vacate API to properly accept and return `integrations_server` to be correct. 

Currently the endpoint only knows about `apm`, which is why parts of the code still don't reference `integrations_server`. Before we add this in `cloud-sdk-go` we would have to extend the API to support it. 

> The tracking updates make sense in essence; however, they contradict how everything in the code gets handled to perform the vacate request.

This is because the tracking uses a separate `search-deployments` API. Most if not all APIs outside of the vacate API will actually differentiate between the two kinds. So using the kind returned by the vacate API will not work.
This workaround changes the query to not use kind, as it may contain the wrong data.

So bottom line is that we also have to fix the API.

**Comment by gigerdo** [comment]
Just found another small bug I overlooked: When tracking the change, also the current step is printed out. This also used the kind.

So now I went through all the usages of kind and replaced it with resource-id (2 usages replaced). This should ensure all the checks work correctly for any case.

**Comment by Z3r0Sum** [comment]
> So bottom line is that we also have to fix the API.

I didn't realize the underlying API didn't support `integrations_server` - thanks for clarifying and addressing this with the workaround!
